### ISSUE TITLE
Font extension fetching bypasses CSS font fetching definitions

### ISSUE THREAD

**Issue by jyasskin** [w3c/IFT#69]
I just found https://drafts.csswg.org/css-fonts-4/#fetch-a-font that defines common Fetch parameters for CSS font fetching. The new IFT fetch integration in https://w3c.github.io/IFT/Overview.html#extend-subset doesn't include all of those parameters, which means there will probably be unexpected effects on CORS, and maybe other things. Y'all should figure out how this is supposed to match up.

https://w3c.github.io/IFT/Overview.html#client-state also says that "The client will need to maintain at minimum the following state for each font file", but it's not obvious what a "font file" is. Is it a URL? Probably not, since that would break [storage partitioning](https://github.com/privacycg/storage-partitioning#user-agent-state). It seems like it should be managed by whatever subsystem will eventually call https://drafts.csswg.org/css-fonts-4/#fetch-a-font, but that isn't wired up yet.

@noamr and @svgeesus might be the right people to think about the architecture here.

**Comment by noamr** [comment]
> I just found https://drafts.csswg.org/css-fonts-4/#fetch-a-font that defines common Fetch parameters for CSS font fetching. The new IFT fetch integration in https://w3c.github.io/IFT/Overview.html#extend-subset doesn't include all of those parameters, which means there will probably be unexpected effects on CORS, and maybe other things. Y'all should figure out how this is supposed to match up.
>
 
True, [fetch a font](https://drafts.csswg.org/css-fonts-4/#fetch-a-font) calls [Fetch a style resource](https://drafts.csswg.org/css-values-4/#fetch-a-style-resource), which does a few things the subset fetch doesn't do:
- Fetches the request as anonymous CORS
- Sends the stylesheet URL as referrer
- Reports the fetch to Resource Timing if the stylesheet and its importing stylesheets are same-origin

I think it would be better to use that infrastructure in case we need to add more things in the future that style requests do and apply them to IFT requests.

**Comment by garretrieger** [comment]
My intention is that the parameters I don't explicitly specify a value for would be set according to whatever is normal for a font request on the particular user agent. It's important to note that this spec is not specific to use in a browser, it may end up implemented in things that don't have the concept of a CSS font face. So some of the settings from https://drafts.csswg.org/css-fonts-4/#fetch-a-font may not all apply (ie. referrer being the style sheet) 

- I'll add some text to the spec making it clear that any request/url parameters that are not specified should be set according to normal behaviour for a font load in that user agent, and I can specifically call out https://drafts.csswg.org/css-fonts-4/#fetch-a-font for platforms that implement it.
- I think I'll also add the cors mode setting of "cors" which is currently unspecified in the current text.

**Comment by jyasskin** [comment]
I wonder if it'd work better to have the wrapper context pass in the fetching algorithm, the various bits of client state, and the codepoints it has discovered it needs... Then the appropriate CSS spec could call into IFT with the right Fetch wrapper and state with appropriate lifetimes, and other contexts that aren't CSS wouldn't be bothered by extra text about it.

**Comment by garretrieger** [comment]
This is a good idea, I just started a PR (#98) which makes the fetching algorithm an input to the subset extension algorithm. It also defines an explicit set of inputs and outputs for that algorithm.
